fastlane/lanes/localization.rb

```diff
@@ -127,55 +129,58 @@
-        gutenberg_path = custom_gutenber_path
-        UI.message("Using Gutenberg from #{gutenberg_path} instead of cloning it...")
-      else
-        # On top of fetching the latest Pods, we also need to fetch the source for the Gutenberg code.
-        # To get it, we need to manually clone the repo, since Gutenberg is distributed via XCFramework.
-        # XCFrameworks are binary targets and cannot extract strings via genstrings from there.
-        config = gutenberg_config!
-
-        ref_node = config[:ref]
-        UI.user_error!('Could not find Gutenberg ref to clone the repository in order to access its strings.') if ref_node.nil?
-
-        ref = ref_node[:tag] || ref_node[:commit]
-        UI.user_error!('The ref to clone Gutenberg in order to access its strings has neither tag nor commit values.') if ref.nil?
-
-        github_org = config[:github_org]
-        UI.user_error!('Could not find GitHub organization name to clone Gutenberg in order to access its strings.') if github_org.nil?
-
-        repo_name = config[:repo_name]
-        UI.user_error!('Could not find GitHub repository name to clone Gutenberg in order to access its strings.') if repo_name.nil?
-
-        # Create a temporary directory to clone Gutenberg into.
-        # We'll run the rest of the automation from within the block, but notice that only the Gutenbreg cloning happens within the temporary directory.
+    # Fetch fresh pods to read the latest localizations from them.
+    # In CI, we expect the pods to be already available and up to date.
+    cocoapods unless is_ci
+
+    # For the same reason, fetch fresh packages.
+    # However, notice we currently need to do this in CI as well.
+    # That's because we haven't yet implemented a method to share the derived data folder explicitly between the CI SPM caching logic and this lane.
+    #
+    # See also:
+    # - https://github.com/wordpress-mobile/WordPress-iOS/pull/23506/files/f694d2c3f433d616deeb15a61a3e9a01f55bc07e#r1725934262
+    # - https://github.com/wordpress-mobile/WordPress-iOS/pull/23506#discussion_r1727419190
+    derived_data_path = options.fetch(:derived_data_path, DERIVED_DATA_PATH)
+    resolve_packages(derived_data_path: derived_data_path)
+
+    custom_gutenberg_path = options[:gutenberg_path]
+    if custom_gutenberg_path
+      # It's simpler to ask the caller to give an absolute path than to make it absolute ourselves
+      # Given this is a debug option, it seems like an okay tradeoff.
+      UI.user_error!("Please provide gutenberg_path as an absolute path. Current value: #{custom_gutenberg_path}") unless File.absolute_path?(custom_gutenberg_path)
+
+      UI.user_error!("Could not find Gutenber project at given path #{custom_gutenberg_path}") unless Dir.exist?(custom_gutenberg_path)
+
+      UI.message("Using Gutenberg from #{custom_gutenberg_path} instead of cloning it...")
+      generate_strings_file(gutenberg_path: custom_gutenberg_path, derived_data_path: derived_data_path)
```

Comment:
I had to extract the call to `genstrings` into this util method in order to create the temp dir only when necessary.

If the structure had been

```
if custom_gutenberg_path
  ...
  path = custom_gutenberg_path
else
  Dir.mktmpdir do |tempdir|
    Dir.chdir(tempdir)
      # checkout gutenberg-mobile
    end
    path = # gutenberg-mobile in tempdir path
end

# call to genstrings with path
```

then in the branch where we did the Git clone, `tempdir` would have been deleted by the time we called `genstrings`.

This solution duplicates the `genstrings` call in both if branches, so that it can run while `tempdir` still exists in the branch with the Git clone. The duplication is minimal, thanks to having extracted the call into `generate_strings_file(gutenberg_path:, derived_data_path:)`

---

I checked to see if Ruby has something like Swift's `defer`, but I couldn't find it. The closest is (`begin ensure end`](https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/ensure.html), but that have defeated the purpose of only created the temp dir when needed.